libs/x-content/src/test/java/org/elasticsearch/xcontent/TextTests.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.xcontent;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.nio.charset.StandardCharsets;
+
+public class TextTests extends ESTestCase {
```

Comment:
🎉

Reply:
This is a pretty good test suite!

Optional: Perhaps consider some tests that use randomization to do a sequence of operations (like `string()`, `stringLength()`, `bytes()`) and assert that the result is right?

This is probably not necessary, since you've already added regression tests for the sequences that bit us before.